lib/util/kivaloo.c

```diff
@@ -0,0 +1,89 @@
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "sock.h"
+#include "warnp.h"
+#include "wire.h"
+
+#include "kivaloo.h"
+
+struct kivaloo_cookie {
```

Comment:
At first glance, I don't see anything kivaloo-specific in here.  Couldn't it be used for `lbs` as well?  I'd be tempted to:
- rename it to something like `wire_wrapper` (if not include it directly in `lib/wire/wire.h`).
- use it in programs like `tests/lbs/` and `tests/kvlds`.